From 56aa51b2fe924b311041e55217d034205f192381 Mon Sep 17 00:00:00 2001
From: Aqeeb Khan <aqeebx.khan@intel.com>
Date: Tue, 4 Jul 2023 08:35:28 +0000
Subject: [PATCH] Add redfish events BMC secure FW update

Added new redfish events
  If ROT-P is corrupted
  If BMCSecureBootReadFailed
  If CopyROTPToROTSFailed
BMCSecureBootDisabled event status as OK.

Tested:
Redfish events are logged successfully and verified
the same in Redfish interface (Postman tool).
BMCSecureBootReadFailed redfish log: Failed to read secure boot status.
ROTPCorrupted redfish log: ROT-P is corrupted. Booted with ROT-S.
CopyROTPToROTSFailed redfish log: Failed to copy ROT-P to ROT-S

Signed-off-by: Gade-Rajasekhar Reddy <raja.sekhar.reddy.gade@intel.com>
Signed-off-by: Aqeeb Khan <aqeebx.khan@intel.com>

%% original patch: 0009-Add-redfish-events-BMC-secure-FW-update.patch
Upstream-Status: Pending
---
 .../registries/openbmc_message_registry.hpp   | 27 +++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/redfish-core/include/registries/openbmc_message_registry.hpp b/redfish-core/include/registries/openbmc_message_registry.hpp
index da41bc6b..ab7f6851 100644
--- a/redfish-core/include/registries/openbmc_message_registry.hpp
+++ b/redfish-core/include/registries/openbmc_message_registry.hpp
@@ -696,6 +696,24 @@ constexpr std::array registry = {
             {"string"},
             "None.",
         }},
+    MessageEntry{"ROTPCorrupted",
+        {
+            "Indicates ROT-P is corrupted. Booted with ROT-S",
+            "%1",
+            "Critical",
+            1,
+            {"string"},
+            "ROT BMC firmware update should be done to recover ROT-P",
+        }},
+    MessageEntry{"CopyROTPToROTSFailed",
+        {
+            "Indicates Failed to copy ROT-P to ROT-S",
+            "%1",
+            "Critical",
+            1,
+            {"string"},
+            "Reset the BMC to retry the copy operation. If failure still persist, update ROT FW.",
+        }},
     MessageEntry{"BMCSecureBootEnabled",
         {
             "Indicates BMC secure boot is enabled.",
@@ -709,6 +727,15 @@ constexpr std::array registry = {
         {
             "Indicates BMC secure boot is disabled.",
             "BMC secure boot is disabled.",
+            "OK",
+            0,
+            {},
+            "None.",
+        }},
+    MessageEntry{"BMCSecureBootReadFailed",
+        {
+            "Indicates Failed to read secure boot status.",
+            "Failed to read secure boot status.",
             "Critical",
             0,
             {},
-- 
2.25.1

